### PR TITLE
Writing Partitioned Datasets recipe for Python

### LINKED ISSUE #46: [Python] Add a recipe on how to save partitioned datasets to the Cookbook
We show how to read paritioned data, but we don't show how to write it in the Cookbook

### PR DESCRIPTION
**Pull request by amol-**
Fixes #46

### CODE REVIEW

File: python/source/io.rst [hunk 1]
```diff
@@ -217,6 +217,63 @@ provided to :func:`pyarrow.csv.read_csv` to drive
...
+
+Arrow will partition datasets in subdirectories by default, which will
+result in 10 different directories named with the value of the partitioning
+column and with file containing the data partition inside:
+
+.. testcode::
+
+    from pyarrow import fs
+
+    s3 = fs.LocalFileSystem()
```
Comment (jorisvandenbossche) [inline python/source/io.rst:257]:
```suggestion
    localfs = fs.LocalFileSystem()
```

? (or don't import `fs`, and use `fs` as the name for the variable)

**Review by jorisvandenbossche — COMMENTED** [review]
I understand that you are using the dummy "chunk" column to be able to have a partitioned dataset without having a natural column to partition on, but my feeling is that this is a bit of a workaround because we don't give a better alternative (yet). 

Personally, I would maybe use a more useful partitioning column as the main example, and then mention the "chunk" column as a way to do it if there is no column already present that can be used for partitioning (and then we can easily update this when a better API becomes available)

File: python/source/io.rst (hunk 1 above)
Comment (amol-) [reply python/source/io.rst]: ops, thanks for catching this one. I copy/pasted another block of code and ended up with the old variable name

**Comment by amol-** [comment]
> I understand that you are using the dummy "chunk" column to be able to have a partitioned dataset without having a natural column to partition on, but my feeling is that this is a bit of a workaround because we don't give a better alternative (yet).
> 
> Personally, I would maybe use a more useful partitioning column as the main example, and then mention the "chunk" column as a way to do it if there is no column already present that can be used for partitioning (and then we can easily update this when a better API becomes available)

Switched to a real column (`year` of birthday) instead of using the chunk column.

**Comment by amol-** [comment]
@jorisvandenbossche can you re-review? Should have addressed your concerns

File: python/source/io.rst [hunk 2]
```diff
@@ -217,6 +217,65 @@ provided to :func:`pyarrow.csv.read_csv` to drive
...
+
+    import pyarrow as pa
+    import pyarrow.dataset as ds
+
+    ds.write_dataset(data, "./partitioned", format="parquet",
+                     partitioning=ds.partitioning(pa.schema([("year", pa.int16())])))
+
+Arrow will partition datasets in subdirectories by default, which will
+result in 10 different directories named with the value of the partitioning
+column and with file containing the data partition inside:
```
Comment (jorisvandenbossche) [inline python/source/io.rst:253]: the "and with file containing the data partition inside" reads a bit strange. Maybe something like "each with a file containing the subset of the data for that partition"

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! One small language remark

File: python/source/io.rst (hunk 2 above)
Comment (amol-) [reply python/source/io.rst]: :+1: reworded

**Review by jorisvandenbossche — APPROVED** [review]
Ready for merge?

**Pull request merged**